generators/server/index.js

```diff
@@ -205,9 +209,12 @@ module.exports = JhipsterServerGenerator.extend({
     prompting: {
 
         askForModuleName: function () {
-            if (this.baseName) return;
+            //need this for having a concrete name in gateway app to build urls like "/jhipster-uaa/api/account"
+            // @TODO read this application name from gateway during its generation
```

Comment:
may be you can ask a question to the user to enter the gateways name here when `uaa` is selected